### PR TITLE
Improve error handling in encryption pkg

### PR DESCRIPTION
**Pull request by kaloyan-raev**
Fixes https://storjlabs.atlassian.net/browse/V3-542

### CODE REVIEW

File: pkg/encryption/aesgcm.go [hunk 1]
```diff
@@ -36,14 +35,14 @@ type aesgcmEncrypter struct {
...
+		return nil, Error.Wrap(err)
 	}
 	aesgcmEncrypt, err := cipher.NewGCM(block)
 	if err != nil {
-		return nil, err
+		return nil, Error.Wrap(err)
 	}
 	if encryptedBlockSize <= aesgcmEncrypt.Overhead() {
-		return nil, Error.New("block size too small")
+		return nil, EncryptedBlockSizeTooSmallError.New(strconv.Itoa(encryptedBlockSize))
```
Comment (coyle) [inline pkg/encryption/aesgcm.go]: This sounds like a Java error :)

Comment (kaloyan-raev) [reply pkg/encryption/aesgcm.go]: Any suggestion how to improve it?

Comment (egonelbre) [reply pkg/encryption/aesgcm.go]: `ErrInvalidParams.New("encrypted block size %d too small", encryptedBlockSize)`. We only need specific error class when we need to distinguish and take action on the specific situation.

Comment (kaloyan-raev) [reply pkg/encryption/aesgcm.go]: I pushed another commit? How is it now?

I intentionally left `ErrDecryptFailed` as a separate class, so I can have a more user-friendly error message when decryption fails. This was the main motivation for this PR.

Comment (coyle) [reply pkg/encryption/aesgcm.go]: I was making a joke but I like the new error type better.

**Pull request merged**